inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/pooling_gpu_ref.cl

```diff
@@ -116,20 +123,34 @@ KERNEL(pooling_gpu)(
 #if OUTPUT_DIMS == 5
     for(uint l = 0; l < POOL_SIZE_Z; l++)
     {
-        int input_offset_z = offset_z + l;
+        #if MAX_POOLING && defined(DILATION_SIZE_Z)
+            int input_offset_z = offset_z + (l * DILATION_SIZE_Z);
```

Comment:
if you define `DILATION_SIZE_*` to `1` when they are not used, you can get rid of `#ifdef` dances here, compiler will optimize away multiplication by `1`